main.js

```diff
@@ -1,60 +1,59 @@
-  {text: 'You\’ve got to ask yourself one question: \‘Do I feel lucky?\’ Well, do ya, punk?', movie: 'Dirty Harry'},
-  {text: 'Hold on to your butts.', movie: 'Jurassic Park'},
-  {text: 'As far back as I can remember, I always wanted to be a gangster.', movie: 'Goodfellas'},
-  {text: 'Human sacrifice! Dogs and cats living together. Mass hysteria!', movie: 'Ghostbusters'}
-];
-const displayArea = document.getElementById('display-area');
-
-bindEvents();
-displayNewQuote();
-
-function bindEvents() {
-  const newQuoteButton = document.getElementById('new-quote-button');
-  newQuoteButton.addEventListener('click', handleNewQuoteClick);
-}
-
-function displayNewQuote() {
-  const quotePara = document.getElementById('quote');
-  const refCite = document.getElementById('ref');
-  const quote = getRandomQuote(quotes);
-
-  quotePara.innerText = '\“' + quote['text'] + '\”';
-  refCite.innerText = quote['movie'];
-  displayArea.classList.toggle('show');
-  updateTwitterIntent(quote);
-};
-
-function getRandomIndex(array) {
-  return Math.floor(Math.random() * array.length);
-}
-
-function getRandomQuote(quotes) {
-  const index = getRandomIndex(quotes);
-  return quotes[index];
-}
-
-function handleNewQuoteClick() {
-  displayArea.classList.toggle('show');
-  setTimeout(displayNewQuote, 350);
-}
-
-function updateTwitterIntent(quote) {
-  const twitterLink = document.getElementById('tweet-quote-link');
-  const intentAddress = `https://twitter.com/intent/tweet?text=“${quote.text}”  — ${quote.movie}`;
-  twitterLink.setAttribute('href', intentAddress);
+fetch('quotes.json')
+  .then(response => {
+    if (!response.ok) {
+      document.getElementById('quote-box').textContent = `
+        Network request failed with status 
+        ${response.status}: ${response.statusText}. 
+        Please refresh to try again.
+      `;
+    }
+    return response.json();
+  })
+  .then(json => {
+    const quotes = json;
+    init(quotes);
+  });
+
+function init(quotes) {
+  const displayArea = document.getElementById('display-area');
+  bindEvents();
+  displayNewQuote();
+
+  function bindEvents() {
+    const newQuoteButton = document.getElementById('new-quote-button');
+    newQuoteButton.addEventListener('click', handleNewQuoteClick);
+  }
+
+  function displayNewQuote() {
+    const quotePara = document.getElementById('quote');
+    const refCite = document.getElementById('ref');
+    const refYear = document.getElementById('ref-year');
+    const quote = getRandomQuote(quotes);
+
+    quotePara.innerText = `“${quote.text}”`;
+    refCite.innerText = quote.movie;
+    refYear.innerText = ` (${quote.year})`;
+    displayArea.classList.toggle('show');
+    updateTwitterIntent(quote);
+  };
+
+  function getRandomIndex(array) {
+    return Math.floor(Math.random() * array.length);
+  }
+
+  function getRandomQuote(quotes) {
+    const index = getRandomIndex(quotes);
```

Comment:
mmm, I like the breaking up of functions above. however, I think we are getting a little too detailed here. usually there is two reasons for using a separate function: you need to reuse it or you want to show a new semantic block. I think `getRandomIndex` and `getRandomQuote` may even be together (only my opinion). We need to ask ourselves: does the breaking up improve readability more than the extra boilerplate (function definition) harms readability.